File: src/main/kotlin/com/example/security/repository/UserRepository.java
package com.example.security.repository;

import com.example.security.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {

    @Query("""
    select u from User u WHERE u.name = :name
    """)
    Optional<User> findByName(String name);

}

File: backend-auth/src/main/kotlin/com/stannis/backendauth/repository/UserRepository.kt
package com.stannis.backendauth.repository

import com.stannis.backendauth.entity.AuthUser
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository: JpaRepository<AuthUser, Long> {

    fun findByLogin(login: String): Optional<AuthUser>

}